Source/Devices/Neuropixels_1.cpp

```diff
@@ -496,8 +515,73 @@ void Neuropixels_1::processFrames()
 		if (shouldAddToBuffer)
 		{
 			shouldAddToBuffer = false;
-			lfpBuffer->addToBuffer(lfpSamples, lfpSampleNumbers, lfpTimestamps, lfpEventCodes, numUltraFrames);
-			apBuffer->addToBuffer(apSamples, apSampleNumbers, apTimestamps, apEventCodes, numUltraFrames * superFramesPerUltraFrame);
+			lfpBuffer->addToBuffer(lfpSamples.data(), lfpSampleNumbers, lfpTimestamps, lfpEventCodes, numUltraFrames);
+			apBuffer->addToBuffer(apSamples.data(), apSampleNumbers, apTimestamps, apEventCodes, numUltraFrames * superFramesPerUltraFrame);
+
+			if (!lfpOffsetCalculated) updateLfpOffsets(lfpSamples, lfpSampleNumbers[0]);
+			if (!apOffsetCalculated) updateApOffsets(apSamples, apSampleNumbers[0]);
+		}
+	}
+}
+
+void Neuropixels_1::updateApOffsets(std::array<float, numApSamples>& samples, int64 sampleNumber)
+{
+	if (sampleNumber > apSampleRate * 5)
+	{
+		uint32_t counter = 0;
+
+		while (apOffsetValues[0].size() <= 99)
```

Comment:
These types of hard coded consts should be declared as `static const` members of the class so that they are changeable in one place if need be and given some semantic context.

Reply:
Examples in this function and those that follow

5
100
99